### PR TITLE
Draft the My Trait Library widget

### PR DESCRIPTION
**Pull request by vlagrassa**
- Re-organizes and modularizes some of the trait querying endpoints
- Modularizes a lot of the Phenotype Database table code
- Uses both of the above to create a rough draft of the MTL widget

(Would prefer to merge #472 into dev and handle conflicts in this branch)

### CODE REVIEW

File: src/modules/site-v2/base/views/api/trait.py [hunk 1]
```diff
@@ -13,27 +24,340 @@
...
+    ],
+    'pagination': {
+      'has_next':     pagination.has_next,
+      'has_prev':     pagination.has_prev,
+      'prev_num':     pagination.prev_num,
+      'next_num':     pagination.next_num,
+      'total_pages':  pagination.pages,
+      'current_page': current_page
+    },
+  }
```
[the remark this answers is not in the document]
Comment (vlagrassa) [reply src/modules/site-v2/base/views/api/trait.py:251]: Values are cleaned using `bleach` in the helper function `get_clean`.

File: src/modules/site-v2/base/views/api/trait.py [hunk 2]
```diff
@@ -13,27 +24,340 @@
...
+
+  # Format return data
+  return {
+    'data': [
+      trait.to_json_with_values() for trait in data
+    ],
+    'draw':            draw,
+    'recordsTotal':    total_records,
+    'recordsFiltered': filtered_records,
+  }
```
[the remark this answers is not in the document]
Comment (vlagrassa) [reply src/modules/site-v2/base/views/api/trait.py:359]: Values are cleaned using `bleach` in the helper function `get_clean`.

File: src/modules/site-v2/templates/_scripts/utils.js [hunk 3]
```diff
@@ -165,3 +165,40 @@ function flash_message(message, full_msg_link=null, full_msg_body=null) {
...
+  else {
+    flash_message(backupMessage);
+  }
+}
+
+
+/* Format a date as YYYY-MM-DD.
+ */
+function formatDate(d) {
+  if(d === null) return 'N/A'
```
Comment (r-vieira) [inline src/modules/site-v2/templates/_scripts/utils.js:194]: consider using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat

**Review by r-vieira — APPROVED** [review]
looks good!

**Pull request merged**